WordPress/Classes/ViewRelated/Reader/ReaderSearchSuggestionsViewController.swift

```diff
@@ -0,0 +1,182 @@
+        let count = suggestionsCount()
+        let numVisibleRows = min(count, MaxTableViewRows)
+        var height = CGFloat(numVisibleRows) * tableView.rowHeight
+        height += ButtonHeight
+        stackViewHeightConstraint.constant = height
+    }
+
+
+    func suggestionsCount() -> Int {
+        return tableViewHandler.resultsController.fetchedObjects?.count ?? 0
+    }
+
+
+    func predicateForFetchRequest() -> NSPredicate? {
+        if phrase.isEmpty {
+            return nil
+        }
+        return NSPredicate(format: "searchPhrase BEGINSWITH[cd] %@", phrase)
+    }
+
+
+    func updatePredicateAndRefresh() {
+        tableViewHandler.resultsController.fetchRequest.predicate = predicateForFetchRequest()
+        do {
+            try tableViewHandler.resultsController.performFetch()
+        } catch let error as NSError {
+            DDLogSwift.logError("Error fetching suggestions after updating the fetch reqeust predicate: \(error.localizedDescription)")
+        }
+        tableView.reloadData()
+    }
+
+
+    // MARK: - Actions
+
+
+    @IBAction func handleClearButtonTapped(sender: UIButton) {
+        let service = ReaderSearchSuggestionService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+        service.deleteAllSuggestions()
+        tableView.reloadData()
+        updateHeightConstraint()
+    }
+}
+
+
+extension ReaderSearchSuggestionsViewController : WPTableViewHandlerDelegate
+{
+    func managedObjectContext() -> NSManagedObjectContext! {
+        return ContextManager.sharedInstance().mainContext
+    }
+
+
+    func fetchRequest() -> NSFetchRequest! {
+        let request = NSFetchRequest(entityName: "ReaderSearchSuggestion")
+        request.predicate = predicateForFetchRequest()
+        request.sortDescriptors = [NSSortDescriptor(key: "date", ascending: false)]
+        return request
+    }
+
+
+    func configureCell(cell: UITableViewCell!, atIndexPath indexPath: NSIndexPath!) {
+        let suggestions = tableViewHandler.resultsController.fetchedObjects as! [ReaderSearchSuggestion]
+        let suggestion = suggestions[indexPath.row]
+        cell.textLabel?.text = suggestion.searchPhrase
+        cell.textLabel?.textColor = WPStyleGuide.darkGrey()
+    }
+
+
+    func tableViewDidChangeContent(tableView: UITableView!) {
+        updateHeightConstraint()
+    }
+
+
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCellWithIdentifier(cellIdentifier, forIndexPath: indexPath)
+        configureCell(cell, atIndexPath: indexPath)
+        return cell
+    }
+
+
+    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        tableView.deselectRowAtIndexPath(indexPath, animated: true)
+
+        let suggestions = tableViewHandler.resultsController.fetchedObjects as! [ReaderSearchSuggestion]
+        let suggestion = suggestions[indexPath.row]
```

Comment:
Please, check out [this puppy](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Extensions/NSFetchedResultsController+Helpers.swift#L8). 

A bit safer (we had few out of bounds crashes ... elsewhere, and i never figured out why!!), but other than that, "type safety" for free 😄 

```
guard let suggestion = tableViewHandler.resultsController.objectOfType(ReaderSearchSuggestion.self, atIndexPath: indexPath) else {
   return
}
```